packages/react/toggle-group/src/ToggleGroup.tsx

```diff
@@ -0,0 +1,320 @@
+
+type ToggleGroupPrimitive = Polymorphic.ForwardRefComponent<
+  | Polymorphic.IntrinsicElement<typeof ToggleGroupSingle>
+  | Polymorphic.IntrinsicElement<typeof ToggleGroupMultiple>,
+  ToggleGroupOwnProps
+>;
+
+const ToggleGroup = React.forwardRef((props, forwardedRef) => {
+  if (props.type === 'single') {
+    return <ToggleGroupSingle {...props} ref={forwardedRef} />;
+  }
+
+  if (props.type === 'multiple') {
+    return <ToggleGroupMultiple {...props} ref={forwardedRef} />;
+  }
+
+  throw new Error(`Missing prop \`type\` expected on \`${TOGGLE_GROUP_NAME}\``);
+}) as ToggleGroupPrimitive;
+
+ToggleGroup.displayName = TOGGLE_GROUP_NAME;
+
+/* -----------------------------------------------------------------------------------------------*/
+
+type ToggleGroupValueContextValue = {
+  value: string[];
+  onItemActivate(value: string): void;
+  onItemDeactivate(value: string): void;
+};
+
+const [
+  ToggleGroupValueProvider,
+  useToggleGroupValueContext,
+] = createContext<ToggleGroupValueContextValue>(TOGGLE_GROUP_NAME);
+
+type ToggleGroupSingleOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof ToggleGroupImpl>,
+  {
+    /**
+     * Allow only one button to be toggled at a time.
+     */
+    type: 'single';
+    /**
+     * The controlled stateful value of the button that is toggled.
+     */
+    value?: string;
+    /**
+     * The value of the button that is toggled when initially rendered. Use
+     * `defaultValue` if you do not need to control the state of a toggle group.
+     */
+    defaultValue?: string;
+    /**
+     * The callback that fires when the value of the toggle group changes.
+     */
+    onValueChange?(value: string): void;
+  }
+>;
+
+type ToggleGroupSinglePrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof ToggleGroupImpl>,
+  ToggleGroupSingleOwnProps
+>;
+
+const ToggleGroupSingle = React.forwardRef((props, forwardedRef) => {
+  const {
+    value: valueProp,
+    defaultValue,
+    onValueChange = () => {},
+    ...toggleGroupSingleProps
+  } = props;
+
+  const [value, setValue] = useControllableState({
+    prop: valueProp,
+    defaultProp: defaultValue,
+    onChange: onValueChange,
+  });
+
+  return (
+    <ToggleGroupValueProvider
+      value={value ? [value] : []}
+      onItemActivate={setValue}
+      onItemDeactivate={() => setValue(undefined)}
+    >
+      <ToggleGroupImpl {...toggleGroupSingleProps} ref={forwardedRef} />
+    </ToggleGroupValueProvider>
+  );
+}) as ToggleGroupSinglePrimitive;
+
+type ToggleGroupMultipleOwnProps = Polymorphic.Merge<
+  Polymorphic.OwnProps<typeof ToggleGroupImpl>,
+  {
+    /**
+     * Allow mutltiple items to be open at the same time.
+     */
+    type: 'multiple';
+    /**
+     * The controlled stateful value of the accordion items whose panels are expanded.
+     */
+    value?: string[];
+    /**
+     * The value of the items whose panels are expanded when the accordion is initially rendered. Use
```

Comment:
there's a bunch of copy paste with the word `accordion`. Not gonna list them all but a search should reveal them.